# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: widget/gtk/DMABufLibWrapper.cpp
# Commit: e00739ac75a4
# Full Hash: e00739ac75a4c2ed38c351f81b26d27fe4f30c90
# Author: stransky <stransky@redhat.com>
# Date: 2025-02-07 04:55:58
# Description:
#   Bug 1946536 [Wayland] Ensure RGBA dmabuf formats if Wayland compositor doesn't provide modifiers for it r=emilio
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D237111
# ==============================================================================

diff -r 0205cda1987c -r e00739ac75a4 widget/gtk/DMABufLibWrapper.cpp
--- a/widget/gtk/DMABufLibWrapper.cpp	Thu Feb 06 22:04:26 2025 +0000
+++ b/widget/gtk/DMABufLibWrapper.cpp	Thu Feb 06 22:06:37 2025 +0000
@@ -250,17 +250,18 @@
 #endif
   if (!formats) {
     formats = new DMABufFormats();
-    formats->EnsureBasicFormats();
   }
+  formats->EnsureBasicFormats();
 
-  if (DRMFormat* format = formats->GetFormat(GBM_FORMAT_XRGB8888)) {
-    mFormatRGBX = new DRMFormat(*format);
-    gfxVars::SetDMABufModifiersXRGB(*format->GetModifiers());
-  }
-  if (DRMFormat* format = formats->GetFormat(GBM_FORMAT_ARGB8888)) {
-    mFormatRGBA = new DRMFormat(*format);
-    gfxVars::SetDMABufModifiersARGB(*format->GetModifiers());
-  }
+  DRMFormat* format = formats->GetFormat(GBM_FORMAT_XRGB8888);
+  MOZ_DIAGNOSTIC_ASSERT(format, "Missing GBM_FORMAT_XRGB8888 dmabuf format!");
+  mFormatRGBX = new DRMFormat(*format);
+  gfxVars::SetDMABufModifiersXRGB(*format->GetModifiers());
+
+  format = formats->GetFormat(GBM_FORMAT_ARGB8888);
+  MOZ_DIAGNOSTIC_ASSERT(format, "Missing GBM_FORMAT_ARGB8888 dmabuf format!");
+  mFormatRGBA = new DRMFormat(*format);
+  gfxVars::SetDMABufModifiersARGB(*format->GetModifiers());
 }
 
 void DMABufDevice::GetModifiersFromGfxVars() {
